catalyst-gateway/bin/src/service/api/registration/mod.rs

```diff
@@ -65,24 +69,28 @@ impl RegistrationApi {
            200: OK<Json<VoterRegistration>>,
            404: NotFound,
            500: ServerError,
+           503: ServiceUnavailable,
        } {
-        let voter = pool
-            .event_db
-            .get_voter(
-                &event_id.0.map(Into::into),
-                voting_key.0 .0,
-                *with_delegators,
-            )
-            .await;
-        match voter {
-            Ok(voter) => {
-                match voter.try_into() {
-                    Ok(voter) => T200(OK(Json(voter))),
-                    Err(err) => T500(server_error!("{}", err.to_string())),
-                }
-            },
-            Err(crate::event_db::error::Error::NotFound(_)) => T404(NotFound),
-            Err(err) => T500(server_error!("{}", err.to_string())),
+        if let Some(event_db) = pool.event_db() {
```

Comment:
I am thinking, is it possible to move this check under the middleware implementation, so in every endpoint we can get an already checked `event_db` instance ?

Reply:
Yeah, replacing the `Option<_>` with a `Result<_>` should do as you suggest, and ensure that the `event_db` instance is unavailable if there is a mismatch, but the middleware is still needed for endpoints that don't make use of `event_db`.

Reply:
The issue I see here is that since the return type for our endpoints isn't a `Result`, it's `UniResponse`, so `?` syntax isn't possible.

Reply:
>The issue I see here is that since the return type for our endpoints isn't a Result, it's UniResponse, so ? syntax isn't possible.

😔It's so pity, would be great to have `?` in such places